### DISCUSSION TITLE
Mark folders with differences in Tree Mode

### DISCUSSION THREAD

**Discussion by Anton-V-K** [WinMerge/winmerge#1075] [Q&A]
When comparing folders, it would be very handy to have a mark for folders which have differences, or hide the identical folders from the list.
Is this feature already available? Which option should I check?
I've selected `Full Content` as compare method for folders.

Here is an example:
![WinMerge-FoldersComparison-test](https://user-images.githubusercontent.com/20116984/145260114-8fe104c7-3087-4d78-a8c5-f00824982cc9.png)
The folder `Extension` does have differences:
![WinMerge-FoldersComparison-test-Extension](https://user-images.githubusercontent.com/20116984/145260303-86ba1a86-a394-4c37-b988-e9d08c60f352.png)
But unfortunately it's not obvious when viewing its parent folder.

Even when 'Show Identical Items' it disabled all folders are shown.

So you have to explore each folder to locate all differences.

Workaround: disable 'Tree Mode` so all files are displayed, and then you can turn identical items off.

**Comment by sdottaka** [comment]
Looking at your first screenshot, it looks like the recursive comparison ('Include Subfolders' option) is disabled.
Please enable the 'Include Subfolders' option as shown below. The folders with differences will be colored as shown in the screenshot below.

![image](https://user-images.githubusercontent.com/98126/145304610-98bcda72-c93f-47d3-9c56-5d2c86aa3ff9.png)

![image](https://user-images.githubusercontent.com/98126/145304143-756a2495-58f7-4919-b319-7c7662f28e33.png)

If you invoke WinMerge from the command line, you must specify the `/r` option, otherwise the folders will not be compared recursively.


The following example is a screenshot of the 'Show Identical items' menu item when unchecked.


![image](https://user-images.githubusercontent.com/98126/145304160-514e374d-686c-4740-ba20-716a0c748250.png)

**Reply by Anton-V-K** [reply]
Thanks for the reply.
The 'Include Subfolders' option was enabled (it seems to be ON always, at least in UI).
I was invoking WinMerge from command-line and wasn't aware of the `/r` switch. With it the comparison works as expected.

It seems like a minor issue in UI: the enabled state of the 'Include Subfolders' option is remembered, but is actually ignored in certain cases.

**Comment by Gitoffthelawn** [comment]
I recently created a discussion or issue item because, I too, wasn't applying the `/r` switch and therefore was surprised by the results.  I wonder if this happens to many people, and could benefit from UX improvement.
